### ISSUE TITLE
Honor Source Image Path in Image Cache

### ISSUE THREAD

**Issue by bhsmither** [cubecart/v6#1386]
See: https://forums.cubecart.com/topic/51944-where-are-the-whole-cache-of-imeges-located/

Feature Request: when rendering sized derivatives of source images, keep in play the path to the source image (as specified in `CubeCart_filemanager`.`filepath`) and establish the same path in the `/images/cache/` folder.

This (probably) will prevent identical filename collisions, and enable better management within some (unreasonable) hosting environments - not that `/images/cache/` should ever need external management.

**Comment by Dirty-Butter** [comment]
Now that 6.1.2 allows admin to upload images to folders on the product listing Image tab - there's a good incentive to create/move images to folders. Sounds like maintaining the folder structure in cache would speed things up even more!

**Comment by bhsmither** [comment]
My diagnostics show that this is already the desired behavior.

Use the File Manager, I moved an image from `/` (which is in reality `/images/source/`) to `/test/` (a folder that already exists which is in reality `/images/source/test/`). The database was updated accordingly.

However, in making this "move", the rendered cached images were not affected.

I manually deleted the rendered images from the `/cache/images/` folder.

I then called for the product's page.

The code tracing showed that the path for this image carried over to the function that creates the rendered images if they do not already exist. Thus, I have the set in a matching folder, for example, `/images/cache/test/prod7.##.png`.

So, I think there is nothing to do here.
